### DISCUSSION TITLE
Multiplayer support

### DISCUSSION THREAD

**Discussion by jogibear9988** [GrapesJS/grapesjs#5919] [Q&A]
Have you already thought about Multiplayer support in grapejs (or has it)?
I'm think of implementing in my designer (https://github.com/node-projects/web-component-designer), but it's a hard task (I think), cause also of the posibillity to edit the source code.

Found this interesting blog post about support in figma: https://www.figma.com/de-de/blog/how-figmas-multiplayer-technology-works/

Wanted to know, what are your thoughts about this.

**Comment by artf** [comment]
For sure we'll need it, I actually built already the multiplayer support for another product a few years back (that Figma article was actually my main reference 😅).

The big problem is how we manage the state and the current data structure in GrapesJS that makes this implementation harder but it should definitely be part of our roadmap.

**Comment by jogibear9988** [comment]
I'm not yet really sure on how to implement multiplayer . For example, in my designer you can edit in design mode, source code or split view. If you edit in Source Code, I reload the whole designer, cause everything could be changed. How should I now solve a conflict if someone moved an elment in design view at the same time?
Have you Ideas for this?

**Reply by artf** [reply]
You have to find a way to diff the changes instead of reloading the whole designer

**Comment by hazho** [comment]
you can achieve this, in any of the following ways:
WebRTC
ServerSentEvent (SSE)
interval-based checking (for example each 10s or more(can be configured per/view or per project))
